src/main/java/org/craftercms/engine/service/context/SiteContext.java

```diff
@@ -309,7 +310,16 @@ public GraphQL getGraphQL() {
     }
 
     public boolean isValid() throws CrafterException {
-        return state == State.INITIALIZED && storeService.validate(context);
+
+        try {
+            if (state == State.CREATED) {
+                logger.debug("Waiting for initialization of {}", this);
+                initializationLatch.await();
```

Comment:
I think this should be an `await` with a 5 min timeout, we don't want to leave threads stuck if the initialization takes too long.